garage/tf/algos/dqn.py

```diff
@@ -17,37 +9,67 @@
 
 
 class DQN(OffPolicyRLAlgorithm):
-    """
-    DQN.
-
-    Args:
-        env_spec (garage.envs.env_spec.EnvSpec): Environment specification.
-        max_path_length (int): Maximum path length. The episode will terminate
-            when length of trajectory reaches max_path_length.
-        qf_lr (float): Learning rate for Q-Function.
-        qf_optimizer (tf.Optimizer): Optimizer for Q-Function.
-        discount (float): Discount factor for rewards.
-        name (str): Name of the algorithm.
-        target_network_update_freq (int): Frequency of updating target network.
-        grad_norm_clipping (float): Maximum clipping value for clipping tensor
-            values to a maximum L2-norm. It must be larger than 0. If None,
-            no gradient clipping is done. For detail, see docstring for
-            tf.clip_by_norm.
-        double_q (bool): Bool for using double q-network.
+    """DQN from https://arxiv.org/pdf/1312.5602.pdf.
+
+    Known as Deep Q-Network, it estimates the Q-value function by deep neural
+    networks. It enables Q-Learning to be applied on high complexity
+    environments. To deal with pixel environments, numbers of tricks are
+    usually needed, e.g. skipping frames and stacking frames as single
+    observation.
     """
 
     def __init__(self,
                  env_spec,
+                 policy,
+                 qf,
                  replay_buffer,
+                 exploration_strategy=None,
+                 n_epoch_cycles=20,
+                 min_buffer_size=int(1e4),
+                 buffer_batch_size=64,
+                 rollout_batch_size=1,
+                 n_train_steps=50,
                  max_path_length=None,
                  qf_lr=0.001,
                  qf_optimizer=tf.train.AdamOptimizer,
                  discount=1.0,
-                 name='DQN',
                  target_network_update_freq=5,
                  grad_norm_clipping=None,
                  double_q=False,
-                 **kwargs):
+                 reward_scale=1.,
+                 input_include_goal=False,
+                 smooth_return=True,
+                 name='DQN'):
+        """
+        Args:
```

Comment:
please document the constructor in the class docstring, as mentioned here https://github.com/rlworkgroup/garage/blob/master/CONTRIBUTING.md#documentation